go/libraries/doltcore/sqle/alterschema.go

```diff
@@ -272,6 +275,8 @@ func updateTableWithModifiedColumn(ctx context.Context, tbl *doltdb.Table, oldSc
 			val, ok := r.GetColVal(modifiedCol.Tag)
 			if !ok || val == nil || val == types.NullValue {
 				return true, fmt.Errorf("cannot change column to NOT NULL when one or more values is NULL")
+			} else if err = validateSpatialTypeSRID(modifiedCol, val); err != nil {
```

Comment:
Should this validation check be inside the `if !modifiedCol.IsNullable()` block? 

It seems like we only need to run this validation when a column type is changing, and you've got that covered in the if block right above this. 

This block seems like it's specific for checking that a modified column that does not allow nulls doesn't have any existing nulls in the data. I'm not sure we need to do anything for SRID here, but let me know if I'm missing something.